src/external/ACSAdapter/AMSFileScanner.ts

```diff
@@ -0,0 +1,168 @@
+import FileMetadata from "@microsoft/omnichannel-amsclient/lib/FileMetadata";
+import FramedClient from "@microsoft/omnichannel-amsclient/lib/FramedClient";
+import WebUtils from "../../utils/WebUtils";
+import { AMSViewScanStatus } from "./AMSFileManager";
+import activityUtils from "./activityUtils";
+
+interface FileScanResponse {
+    status: string;
+}
+
+interface FileScanResult {
+    fileMetadata: FileMetadata;
+    scan: FileScanResponse;
+    next?: any; // eslint-disable-line @typescript-eslint/no-explicit-any
+    activity?: any; // eslint-disable-line @typescript-eslint/no-explicit-any
+}
+
+const defaultScanPollingInterval = 7 * 1000;
+const defaultScanStatusRetrievalDelay = 1000;
+
+interface AMSFileScannerOptions {
+    pollingInterval?: number;
+    scanStatusRetrievalDelay?: number;
+}
+
+class AMSFileScanner {
+    public amsClient: FramedClient;
+    public scanResults: Map<string, FileScanResult> | null = null;
+    private shouldQueueScan: boolean;
+    private test: boolean;
+    private options: AMSFileScannerOptions;
+
+    constructor(amsClient: FramedClient, options: AMSFileScannerOptions = {}) {
+        this.amsClient = amsClient;
+        this.scanResults = new Map<string, FileScanResult>();
+        this.shouldQueueScan = false;
+        this.test = false;
+        this.options = options;
+        this.queueScan();
+    }
+
+    public async queueScan(): Promise<void> {
+        this.shouldQueueScan = true;
+        try {
+            await this.scanFiles();
+        } catch (e) {
+            console.error(e);

```

Comment:
could we add more descriptive error?, at least some sort of hint that tell us where the error was originated for researching failure scenarios